agent/structs/config_entry_intentions.go

```diff
@@ -536,168 +577,256 @@ func (e *ServiceIntentionsConfigEntry) HasAnyPermissions() bool {
 }
 
 func (e *ServiceIntentionsConfigEntry) validate(legacyWrite bool) error {
+
 	if e.Name == "" {
-		return fmt.Errorf("Name is required")
+		return &IntentionValidationError{
+			Position: "Destination",
+			Field: "Name",
+			Violation: "service name is required and cannot be empty",
+		}
 	}
 
 	if err := validateIntentionWildcards(e.Name, &e.EnterpriseMeta); err != nil {
-		return err
+		// Add Position information (which can't be added by validateIntentionWildcards)
+		return AddPositionToIntentionValidationError(err, "Destination")
 	}
 
 	destIsWild := e.HasWildcardDestination()
 
 	if legacyWrite {
 		if len(e.Meta) > 0 {
-			return fmt.Errorf("Meta must be omitted for legacy intention writes")
+			return &IntentionValidationError{
+				Position: "Destination",
+				Field: "Meta",
+				Violation: "must be omitted for legacy intention writes",
+			}
 		}
 	} else {
 		if err := validateConfigEntryMeta(e.Meta); err != nil {
-			return err
+			return &IntentionValidationError{
+				Position: "Destination",
+				Field: "Meta",
+				Violation: err.Error(),
+			}
 		}
 	}
 
 	if len(e.Sources) == 0 {
-		return fmt.Errorf("At least one source is required")
+			return &IntentionValidationError{
+				Position: "Sources",
+				Violation: "at least one source is required",
+			}
 	}
 
 	seenSources := make(map[ServiceName]struct{})
 	for i, src := range e.Sources {
+		position := fmt.Sprintf("Sources[%d]", i)
+
 		if src.Name == "" {
-			return fmt.Errorf("Sources[%d].Name is required", i)
+			return &IntentionValidationError{
+				Position: position,
+				Field: "Name",
+				Violation: "field is required and cannot be empty",
+			}
 		}
 
 		if err := validateIntentionWildcards(src.Name, &src.EnterpriseMeta); err != nil {
-			return fmt.Errorf("Sources[%d].%v", i, err)
+			// Add Position information (which can't be added by validateIntentionWildcards)
+			return AddPositionToIntentionValidationError(err, position)
 		}
 
 		if err := validateSourceIntentionEnterpriseMeta(&src.EnterpriseMeta, &e.EnterpriseMeta); err != nil {
-			return fmt.Errorf("Sources[%d].%v", i, err)
+			return fmt.Errorf("%s.%v", position, err)
```

Comment:
@kisunji : I didn't touch this error message (other than re-using `position` rather than recreating it from `Sources[%d]`) because I hadn't investigated what `validateSourceIntentionEnterpriseMeta` can return in enterprise (and how it would fit into this new framework for generating intention validation errors). Let me know if I need to look into that.